### PR TITLE
feat(module:float-button): add float-button component

### PR DESCRIPTION
**Pull request by OriginRing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by NRaf** [comment]
Hi,

Just came across this feature in the React branch and noticed it's missing from ng-zorro. Any idea on when this will be merged in?

File: components/float-button/demo/group-menu.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+order: 5
+title:
+  zh-CN: 菜单模式
+  en-US: Menu mode
+---
+
+## zh-CN
+
+设置 `nzTrigger` 属性即可开启菜单模式。提供 hover 和 click 两种触发方式。
```
Comment (Laffery) [inline components/float-button/demo/group-menu.md:10]:
```suggestion
设置 `nzTrigger` 属性即可开启菜单模式。提供 `hover` 和 `click` 两种触发方式。
```

File: components/float-button/demo/group.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  en-US: FloatButton Group
+---
+
+## zh-CN
+
+按钮组合使用时，推荐使用 `nz-float-button-group`，并通过设置 `nzShape` 属性改变悬浮按钮组的形状。
+
+## en-US
+
+When multiple buttons are used together, `nz-float-button-group` is recommended. By setting `nzShape` of FloatButtonGroup, you can change the shape of group.
```
Comment (Laffery) [inline components/float-button/demo/group.md:14]: FloatButtonGroup 是 react 版中的组件， 这里表述应为 zorro 中的组件

File: components/float-button/doc/index.en-US.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+---
+category: Components
+type: General
+title: FloatButton
+cover: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*HS-wTIIwu0kAAAAAAAAAAAAADrJ8AQ/original
+tag: 18.1.0
```
Comment (Laffery) [inline components/float-button/doc/index.en-US.md:6]: 18.2.0

File: components/float-button/float-button-content.component.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            <span nz-icon nzType="file-text" nzTheme="outline"></span>
+          </div>
+        }
+      </div>
+    </div>
+  `
+})
+export class NzFloatButtonContentComponent {
+  @Input() nzIcon: TemplateRef<void> | null = null;
+  @Input() nzDescription: TemplateRef<void> | null = null;
```
Comment (Laffery) [inline components/float-button/float-button-content.component.ts:42]: 支持直接传 string 吗

File: components/float-button/float-button-group.component.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  @Input() nzShape: 'circle' | 'square' = 'circle';
+  @Input() nzTrigger: 'click' | 'hover' | null = null;
+  @Input() nzOpen: boolean | null = null;
+  @Output() readonly nzOnOpenChange = new EventEmitter<boolean>();
+  isOpen: boolean = false;
+  dir: Direction = 'ltr';
+
+  constructor(
+    private destroy$: NzDestroyService,
+    @Optional() private directionality: Directionality,
```
Comment (Laffery) [inline components/float-button/float-button-group.component.ts:93]: 写法上参考 #8567 已经将所有可选写法都改成用 inject 的方式注入

File: components/float-button/float-button.component.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  ngOnInit(): void {
+    this.directionality.change?.pipe(takeUntil(this.destroy$)).subscribe((direction: Direction) => {
+      this.dir = direction;
+      this.cdr.detectChanges();
+    });
+
+    this.dir = this.directionality.value;
+  }
+
+  detectChanges(): void {
```
Comment (Laffery) [inline components/float-button/float-button.component.ts:100]: 这个方法没看到用处

**Review by Laffery — CHANGES_REQUESTED** [review]

**Review by Laffery — APPROVED** [review]
LGTM

File: scripts/site/template/ng-zorro-antd.module.template.ts
Comment (Laffery) [inline scripts/site/template/ng-zorro-antd.module.template.ts:1]: Please rebase from master, this file had been removed

File: components/float-button/doc/index.zh-CN.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+---
+category: Components
+subtitle: 悬浮按钮
+type: 通用
+title: FloatButton
+cover: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*HS-wTIIwu0kAAAAAAAAAAAAADrJ8AQ/original
+tag: 18.2.0
```
Comment (Laffery) [inline components/float-button/doc/index.zh-CN.md:7]:
```suggestion
tag: 19.0.0
```

File: components/float-button/doc/index.en-US.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
+---
+category: Components
+type: General
+title: FloatButton
+cover: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*HS-wTIIwu0kAAAAAAAAAAAAADrJ8AQ/original
+tag: 18.2.0
```
Comment (Laffery) [inline components/float-button/doc/index.en-US.md:6]:
```suggestion
tag: 19.0.0
```

**Review by Laffery — CHANGES_REQUESTED** [review]

File: components/back-top/doc/index.en-US.md [hunk 9]
```diff
@@ -5,6 +5,8 @@ title: BackTop
 cover: https://gw.alipayobjects.com/zos/alicdn/tJZ5jbTwX/BackTop.svg
 ---
 
+* Note: This component will be removed later. Please use `<nz-float-button-top></nz-float-button-top>` in the FloatButton component.
```
Comment (lppedd) [inline components/back-top/doc/index.en-US.md:8]: Just wondering, does that mean the backtop component is to be considered deprecated?  
Is it in preparation for Ant Design 5?

**Pull request merged**